core/ecschema-metadata/src/Interfaces.ts

```diff
@@ -28,67 +28,67 @@ import { Unit } from "./Metadata/Unit";
 import { UnitSystem } from "./Metadata/UnitSystem";
 import { SchemaItemKey, SchemaKey } from "./SchemaKey";
 
-/** @beta */
+/** @public @preview */
 export type LazyLoadedSchema = SchemaKey & DelayedPromise<Schema> & Promise<Schema>;
 
-/** @beta */
+/** @public @preview */
 export type LazyLoadedSchemaItem<T extends SchemaItem> = Readonly<SchemaItemKey> & DelayedPromise<T> & Promise<T>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedECClass = LazyLoadedSchemaItem<ECClass>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedEntityClass = LazyLoadedSchemaItem<EntityClass>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedMixin = LazyLoadedSchemaItem<Mixin>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedStructClass = LazyLoadedSchemaItem<StructClass>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedCustomAttributeClass = LazyLoadedSchemaItem<CustomAttributeClass>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedRelationshipClass = LazyLoadedSchemaItem<RelationshipClass>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedEnumeration = LazyLoadedSchemaItem<Enumeration>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedKindOfQuantity = LazyLoadedSchemaItem<KindOfQuantity>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedPropertyCategory = LazyLoadedSchemaItem<PropertyCategory>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedRelationshipConstraintClass = LazyLoadedSchemaItem<EntityClass | Mixin | RelationshipClass>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedUnit = LazyLoadedSchemaItem<Unit>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedInvertedUnit = LazyLoadedSchemaItem<InvertedUnit>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedConstant = LazyLoadedSchemaItem<Constant>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedPhenomenon = LazyLoadedSchemaItem<Phenomenon>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedUnitSystem = LazyLoadedSchemaItem<UnitSystem>;
-/** @beta */
+/** @public @preview */
 export type LazyLoadedFormat = LazyLoadedSchemaItem<Format>;
```

Comment:
I wonder if we shouldn't change the LazyXX types to be a intersection of the payload & Promise.
Example: ```ts LazyLoadedSchema = SchemaKey & Promise<Schema>; ```

That would also allow us to keep the DelayedPromise internal and an implementation detail (in fact DelayedPromise is more or less a wrapper around a promise)

Reply:
I have not thought through the implications of this but I don't really see anything wrong with your suggestion.  I'd like to hear what other people think

Reply:
I like that change.
BTW I just found out LazyLoadedFormat is not used once. Nowhere in the entire repo.

I would prefer we get rid of all this LazyLoadedXXX stuff. I know it seems nice to call myEntity.baseClass.Name, while myEntity.baseClass is at the same time a promise that can get it. But this is totally overengineered. I did not find a single spot in our current code where this promise is actually a promise. I'll have a call with Stefan to better understand if our incremental loading makes use of this...

Reply:
I removed the DelayedPromises from the LazyLoaded definition and tagged the DelayedPromise types internal.
Also added an issue to address that items get properly loaded asynchronously:
https://github.com/iTwin/itwinjs-backlog/issues/1467